WireDomain/Sources/WireDomain/Event Decryption/ProteusMessageDecryptor.swift

```diff
@@ -0,0 +1,152 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireDataModel
+
+// sourcery: AutoMockable
+/// Decrypt proteus messages.
+protocol ProteusMessageDecryptorProtocol {
+
+    /// Decrypt a proteus message.
+    ///
+    /// - Parameter eventData: A payload containing the encrypted message.
+    /// - Returns: The payload containing the decrypted message.
+
+    func decryptedEventData(
+        from eventData: ConversationProteusMessageAddEvent
+    ) async throws -> ConversationProteusMessageAddEvent
+
+}
+
+struct ProteusMessageDecryptor: ProteusMessageDecryptorProtocol {
+
+    let proteusService: any ProteusServiceInterface
+    let managedObjectContext: NSManagedObjectContext
+
+    private let maxCiphertextSize = Int(12_000 * 1.5)
```

Comment:
💅 typo?
```suggestion
    private let maxCipherTextSize = Int(12_000 * 1.5)
```

Reply:
[Ciphertext](https://en.wikipedia.org/wiki/Ciphertext) and Plaintext are both single words, hence I didn't capitalize mid-word.